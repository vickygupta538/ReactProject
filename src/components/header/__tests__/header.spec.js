import React from "react";
import renderer from "react-test-renderer";
import Header from "../index";
import logo from "../../assets/amazon_logo.svg";

const setUp = (props = {}) => {
  const component = renderer.create(<Header />);
  return component;
};

describe("Header Component Test Cases", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should match snapshot renderer", () => {
    const wrapper = component.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render without errors", () => {
    const wrapper = component.root.find(el => el.type === "header" && el.props["data-test"] === "headerComponent");
    expect(wrapper).toBeTruthy();
  });
  it("Should render img tag", () => {
    const wrapper = component.root.find(el => el.type === "img");

    expect(wrapper.props.src).toBe(logo);
  });

  it("Should render heading content", () => {
    const wrapper = component.root.find(el => el.type === "div" && el.props["className"] === "headingContent");
    expect(wrapper.children[0]).toBe("Star Wars Character Search");
  });
});
