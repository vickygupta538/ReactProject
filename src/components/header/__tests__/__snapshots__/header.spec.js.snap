// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test Cases should match snapshot renderer 1`] = `
<header
  className="header"
  data-test="headerComponent"
>
  <div
    className="headingContainer"
  >
    <img
      alt=""
      src={null}
    />
    <div
      className="headingContent"
    >
      Star Wars Character Search
    </div>
  </div>
</header>
`;
